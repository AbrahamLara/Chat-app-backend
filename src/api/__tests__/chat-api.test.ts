import { SuperAgentTest } from 'supertest';
import { Server } from 'http';
import { MOCK_JWT, mockServerAndAgent } from '../../test_utils/mock-utils';
import { mockTokenUtils } from '../../test_utils/file_mocks/mock-token-utils';

let server: Server;
let agent: SuperAgentTest;

mockTokenUtils();

describe('Chat endpoint', () => {
  beforeEach(
    mockServerAndAgent((mockServer, testAgent) => {
      server = mockServer;
      agent = testAgent;
    })
  );

  afterEach(async cb => {
    server.close(cb);
  });

  it('returns error message if payload values are invalid', async () => {
    const response1 = await createChatResponse({ userIDs: [] });

    // Check error for blank userIDs value.
    expect(response1.ok).toEqual(false);
    expect(response1.status).toEqual(400);
    expect(response1.body).toMatchSnapshot();

    const response2 = await createChatResponse({
      userIDs: ['id-1'],
      chatName: '',
    });

    // Check error for blank chat name value.
    expect(response2.ok).toEqual(false);
    expect(response2.status).toEqual(400);
    expect(response2.body).toMatchSnapshot();

    const response3 = await createChatResponse({
      userIDs: ['id'],
      chatName: 'test',
      message: '',
    });

    // Check error for blank message value.
    expect(response3.ok).toEqual(false);
    expect(response3.status).toEqual(400);
    expect(response3.body).toMatchSnapshot();
  });
});

function createChatResponse(payload: Record<string, any>) {
  return agent
    .post(`/api/chat`)
    .send(payload)
    .set('Content-Type', 'application/json')
    .set('Authorization', `Bearer ${MOCK_JWT}`);
}
