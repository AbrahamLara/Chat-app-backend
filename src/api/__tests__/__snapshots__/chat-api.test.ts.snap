// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat endpoint returns error message if payload values are invalid 1`] = `
Object {
  "errors": Array [
    Object {
      "field": "chatName",
      "message": "Chat name must be a non-empty string.",
    },
    Object {
      "field": "message",
      "message": "Chat message must be a non-empty string.",
    },
    Object {
      "field": "userIDs",
      "message": "The provided userIDs value is invalid.",
    },
  ],
  "type": "chat",
}
`;

exports[`Chat endpoint returns error message if payload values are invalid 2`] = `
Object {
  "errors": Array [
    Object {
      "field": "chatName",
      "message": "Chat name must be a non-empty string.",
    },
    Object {
      "field": "message",
      "message": "Chat message must be a non-empty string.",
    },
  ],
  "type": "chat",
}
`;

exports[`Chat endpoint returns error message if payload values are invalid 3`] = `
Object {
  "errors": Array [
    Object {
      "field": "message",
      "message": "Chat message must be a non-empty string.",
    },
  ],
  "type": "chat",
}
`;
